lib/PuppeteerSharp/JSHandle.cs

```diff
@@ -7,23 +7,44 @@
 
 namespace PuppeteerSharp
 {
+    /// <summary>
+    /// JSHandle represents an in-page JavaScript object. JSHandles can be created with the <see cref="Page.EvaluateExpressionHandleAsync(string)"/> and <see cref="Page.EvaluateFunctionHandleAsync(string, object[])"/> methods.
+    /// </summary>
     public class JSHandle
     {
-        private ExecutionContext _context;
-        protected readonly Session _client;
-        protected readonly ILogger _logger;
-
-        public JSHandle(ExecutionContext context, Session client, object remoteObject)
+        internal JSHandle(ExecutionContext context, Session client, object remoteObject)
         {
-            _context = context;
-            _client = client;
-            _logger = _client.Connection.LoggerFactory.CreateLogger(this.GetType());
+            ExecutionContext = context;
+            Client = client;
+            Logger = Client.Connection.LoggerFactory.CreateLogger(this.GetType());
             RemoteObject = remoteObject;
         }
 
-        public ExecutionContext ExecutionContext => _context;
+        /// <summary>
+        /// Gets the execution context.
+        /// </summary>
+        /// <value>The execution context.</value>
+        public ExecutionContext ExecutionContext { get; }
+        /// <summary>
+        /// Gets or sets a value indicating whether this <see cref="PuppeteerSharp.JSHandle"/> is disposed.
+        /// </summary>
+        /// <value><c>true</c> if disposed; otherwise, <c>false</c>.</value>
         public bool Disposed { get; set; }
+        /// <summary>
+        /// Gets or sets the remote object.
+        /// </summary>
+        /// <value>The remote object.</value>
         public dynamic RemoteObject { get; internal set; }
```

Comment:
do we need the `internal set` here?

Reply:
No, we don't. I removed that and I also change the Disposed setter to private.